### PR TITLE
Support for Patmos platform

### PR DESCRIPTION
**Pull request by EhsanKhodadad**

### CODE REVIEW

**Comment by lhstrh** [comment]
@EhsanKhodadad, I enabled CI for you, but it looks like there are still some test failures. Please address those first.

**Comment by EhsanKhodadad** [comment]
> @EhsanKhodadad, I enabled CI for you, but it looks like there are still some test failures. Please address those first.

Thank you for enabling CI. The tester has not found a header file in the t-crest repository, because t-crest hasn't cloned on the tester's machine.

**Comment by lhstrh** [comment]
Got it. Can you fix it in the workflow?

**Comment by EhsanKhodadad** [comment]
> Got it. Can you fix it in the workflow?

Sure

**Comment by lhstrh** [comment]
Btw, I noticed tests in your fork were failing. Here's a fix for that: https://github.com/lf-lang/lingua-franca/pull/2402

File: .github/workflows/c-patmos-tests.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Setup and build Patmos
+        uses: ./.github/actions/setup-patmos
+      - name: Check out specific ref of reactor-c
+        uses: actions/checkout@v3
+        with:
+          repository: lf-lang/reactor-c
+          path: core/src/main/resources/lib/c/reactor-c
+          ref: ${{ inputs.runtime-ref }}
+        if: ${{ inputs.runtime-ref }}
+      - name: Run Patmos smoke tests
```
Comment (lhstrh) [inline .github/workflows/c-patmos-tests.yml:48]: This step isn't actually running any tests.

**Comment by EhsanKhodadad** [comment]
I have two questions regarding this PR: 
1. Since I haven't implemented multi-thread functions and variables like mutex and thread in lf-patmos-support yet, can I bypass the tests in the concurrent folder temporarily? If so how?
2. In the regular tests, all LF apps, including my HelloPatmos.lf app is tested at the beginning of the test procedure, whereas the Patmos library, including Patmos-clang, still needs to be installed. Can I postpone their execution after the Patmos installation? These tests are written in .github/workflows/c-test.yml file, while the Patmos installation is invoked by c-patmos-test.yml.

**Comment by EhsanKhodadad** [comment]
> I have two questions regarding this PR:
> 
> 1. Since I haven't implemented multi-thread functions and variables like mutex and thread in lf-patmos-support yet, can I bypass the tests in the concurrent folder temporarily? If so how?
> 2. In the regular tests, all LF apps, including my HelloPatmos.lf app is tested at the beginning of the test procedure, whereas the Patmos library, including Patmos-clang, still needs to be installed. Can I postpone their execution after the Patmos installation? (Something similar to Patmos-tests, where benchmarks are run after Patmos installation.) These tests are written in .github/workflows/c-test.yml file, while the Patmos installation is invoked by c-patmos-test.yml.

@lhstrh @erlingrj @lsk567

**Comment by erlingrj** [comment]
> I have two questions regarding this PR:
> 
>     1. Since I haven't implemented multi-thread functions and variables like mutex and thread in lf-patmos-support yet, can I bypass the tests in the concurrent folder temporarily? If so how?
> 

Check out your file CPatmosTests.java, here you explicitly run these tests with `buildPatmosConcurrent` remove this function and then change the CI so it doesn't try to run the gradle task associated with ut.

>     2. In the regular tests, all LF apps, including my HelloPatmos.lf app is tested at the beginning of the test procedure, whereas the Patmos library, including Patmos-clang, still needs to be installed. Can I postpone their execution after the Patmos installation? (Something similar to Patmos-tests, where benchmarks are run after Patmos installation.) These tests are written in .github/workflows/c-test.yml file, while the Patmos installation is invoked by c-patmos-test.yml.

You should not run any Patmos-specific tests in C/regular-tests. They should be excluded from there and only be built in there special target. Do a search for `ZEPYHR_UNTHREADED` in the code-base to see the additions needed to disable a test category from the regular tests (and also from the CCpp tests)

File: core/src/main/resources/lib/cpp/reactor-cpp
Comment (lhstrh) [inline core/src/main/resources/lib/cpp/reactor-cpp:1]: It doesn't seem like reactor-cpp changes should be included in this PR.

**Review by lhstrh — COMMENTED** [review]
This overall look good! Let me defer to @erlingrj for final approval.

**Review by erlingrj — CHANGES_REQUESTED** [review]
Looks good, but we should not update reactor-cpp submodule in this PR

**Comment by lhstrh** [comment]
@EhsanKhodadad, please make sure reactor-cpp points to [dfdac2c](https://github.com/lf-lang/reactor-cpp/tree/dfdac2c19e8d111cf4741c8bf8f304678a59d025).

**Review by erlingrj — APPROVED** [review]
Looks good, lets merge :)

**Comment by schoeberl** [comment]
Finally, very good!

**Pull request merged**